test/metabase/driver/postgres_test.clj

```diff
@@ -496,53 +489,61 @@
           (jdbc/execute! spec [describe-json-table-sql]))
-        (mt/with-temp Database [database {:engine :postgres, :details details}]
-          (is (= [:type/JSON :type/SerializedJSON]
-                 (->> (sql-jdbc.sync/describe-table :postgres database {:name "describe_json_table"})
-                      (:fields)
-                      (:take 1)
-                      (first)
-                      ((juxt :base-type :semantic-type)))))
-          (is (= '#{{:name              "incoherent_json_val → b",
-                     :database-type     "text",
-                     :base-type         :type/Text,
-                     :database-position 0,
-                     :nfc-path          [:incoherent_json_val "b"]
-                     :visibility-type   :normal}
-                    {:name              "coherent_json_val → a",
-                     :database-type     "bigint",
-                     :base-type         :type/Integer,
-                     :database-position 0,
-                     :nfc-path          [:coherent_json_val "a"]
-                     :visibility-type   :normal}
-                    {:name              "coherent_json_val → b",
-                     :database-type     "bigint",
-                     :base-type         :type/Integer,
-                     :database-position 0,
-                     :nfc-path          [:coherent_json_val "b"]
-                     :visibility-type   :normal}
-                    {:name "coherent_json_val → c",
-                     :database-type     "timestamp",
-                     :base-type         :type/DateTime,
-                     :database-position 0,
-                     :visibility-type   :normal,
-                     :nfc-path          [:coherent_json_val "c"]}
-                    {:name              "incoherent_json_val → c",
-                     :database-type     "double precision",
-                     :base-type         :type/Number,
-                     :database-position 0,
-                     :visibility-type   :normal,
-                     :nfc-path          [:incoherent_json_val "c"]}
-                    {:name              "incoherent_json_val → d",
-                     :database-type     "bigint",
-                     :base-type         :type/Integer,
-                     :database-position 0,
-                     :visibility-type   :normal,
-                     :nfc-path          [:incoherent_json_val "d"]}}
-                 (sql-jdbc.sync/describe-nested-field-columns
-                  :postgres
-                  database
-                  {:name "describe_json_table"}))))))))
+        (mt/with-temp* [Database [database {:engine :postgres, :details details}]]
+          (mt/with-db database
+            (is (= [:type/JSON :type/SerializedJSON]
+                   (->> (sql-jdbc.sync/describe-table :postgres database {:name "describe_json_table"})
+                        (:fields)
+                        (:take 1)
```

Comment:
lol not your original mistake but `(:take 1 <threaded-value>)` is operating as `identity` here as `1` is not associative and therefore `:take` as a function is returning the not-found threaded value. Let's fix this. (also take 1 and first together seem a bit duplicative in the first place)